README.md

```diff
@@ -41,34 +36,27 @@ Add the following step at the end of your job.
     commit_user_email: my-github-actions-bot@example.org
     commit_author: Author <actions@github.com>
 
-    # Optional tag message. 
+    # Optional tag message 
     # Action will create and push a new tag to the remote repository and the defined branch
     tagging_message: 'v1.0.0'
 ```
 
-The Action will only commit files back, if changes are available. The resulting commit **will not trigger** another GitHub Actions Workflow run!
-
-We recommend to use this Action in Workflows, which listen to the `pull_request` event. You can then use the option `branch: ${{ github.head_ref }}` to set up the branch name correctly.
-If you don't pass a branch name, the Action will try to push the commit to a branch with the same name, as with which the repo has been checked out.
-
-## Example Usage
+## Example
 
-The most common use case for this Action is to create a new build of your project on GitHub Actions and commit the compiled files back to the repository.
-Another simple use case is to run a linter and commit the fixes back to the repository.
+In this example, we're running `php-cs-fixer` in a PHP project to fix the codestyle automatically,
+then commit possible changed files back to the repository.
 
-In this example, we're running `php-cs-fixer` in a PHP project, let the linter fix possible code issues and commit the changed files back to the repository.
-
-### Example: Listen to `pull_request` event
-
-When using the Action while listening to the `pull_request`-event, you must add the `ref`-input to the `actions/checkout@v2` step. 
-Otherwhise the repository is checked out in a detached state which causes issue with this Action.
-
-It's also recommended to add the `branch`-input to the `git-auto-commit`-step. This way you tell the Action exactly where to push the commit.
+Note that we explicitly specify `${{ github.head_ref }}` in both the checkout and the commit Action.
+This is required in order to work with the `pull_request` event (or any other non-`push` event).
 
 ```yaml
 name: php-cs-fixer
 
-on: pull_request
+on:
+  pull_request:
+  push:
+    branches:
+      - "master"
```

Comment:
Did this example work? Usually listening to both pull_request and push events causes issues with the Action. 
I assume it worked, as the push event is only triggered on `master`?

Reply:
Just to be sure, i tried this config in a project of mine.

Here is the successful run on master: https://github.com/spawnia/sailor/runs/650648517

And here is what happens on a PR: https://github.com/spawnia/sailor/runs/650660575

Reply:
This is great. Thanks!